### PR TITLE
FIR: set proper visibility of backing fields.

### PR DESCRIPTION
**Pull request by jsjeon**

### CODE REVIEW

File: compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/ClassMemberGenerator.kt [hunk 1]
```diff
@@ -168,6 +168,13 @@ internal class ClassMemberGenerator(
         return irProperty
     }
 
+    private val FirProperty.fieldVisibility: Visibility
+        get() = when {
+            isLateInit -> setter?.visibility ?: status.visibility
+            isConst -> status.visibility
+            else -> Visibilities.PRIVATE
+        }
+
```
Comment (jsjeon) [inline compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/ClassMemberGenerator.kt]: Counterpart I referred is: https://github.com/JetBrains/kotlin/blob/master/compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/PropertyGenerator.kt#L183 which includes computation of JVM field visibility, whereas here we don't deal with JVM fields, right?

Comment (mglukhikh) [reply compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/ClassMemberGenerator.kt]: Well, yes, JVM fields is a separate questions.

**Comment by mglukhikh** [comment]
Merged manually, thank you!

File: compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/ClassMemberGenerator.kt (hunk 1 above)
Comment (jsjeon) [reply compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/ClassMemberGenerator.kt:177]: Addressed at https://github.com/JetBrains/kotlin/pull/3570

**Pull request closed without merging**